worktree_commit.go

```diff
@@ -61,16 +59,40 @@ func (w *Worktree) Commit(msg string, opts *CommitOptions) (plumbing.Hash, error
 		return plumbing.ZeroHash, err
 	}
 
+	allowEmpty := opts != nil && opts.AllowEmptyCommits
```

Comment:
I added this check because of #1051. If you think this is unecessary, I'll remove this line.

Reply:
Nice one. Please add a "Fixes #1051" at the PR description, so that gets closed automatically when we get this merged.

Reply:
The issue will still exist whether I add this line or not.
I think we shouldn't close that issue with this PR because it happens not only in this function.
This line seems redundant if we are not going to solve this issue in this PR.
I think I should remove it in next push.